### DISCUSSION TITLE
rocketmq jvm增加

### DISCUSSION THREAD

**Discussion by jaylu163** [apache/rocketmq#9001] [General]
jvm 内存增加 设置jvm的属性  为xms=4G xmn=4G 运行内存增加

开始运行的时候是几百兆内存。运行一段实际，但是没有写入数据，内存会到2G左右，是否有内存泄漏问题

![image](https://github.com/user-attachments/assets/2d790c4c-da3e-4d07-814b-8ad34428650d)

**Comment by hexueyuan** [comment]
master broker会给MappedFIle提前申请内存并加锁，即使没有请求这个也会有一部分固定内存占用

**Reply by jaylu163** [reply]
我看会一直增加呢，这个默认给4G感觉都不够用，数据量大了，不知道会有问题不

我看namesrv也会有增加内存。

**Reply by jaylu163** [reply]
我理解是用了mmap的技术是吧。那这个namesrv数据查询也有用到，所以增加。但是不确定是否有内存泄漏。一直涨上去

**Reply by hexueyuan** [reply]
MappedFile占用的内存不会一直涨，如果确实在无流量情况下内存一直在增加可以用内存检测工具看看是哪块对象在占用新增内存。
